RecoPixelVertexing/PixelTrackFitting/src/PixelFitterByHelixProjections.cc

```diff
@@ -104,36 +104,20 @@ reco::Track* PixelFitterByHelixProjections::run(
   int nhits = hits.size();
   if (nhits <2) return 0;
 
-  vector<GlobalPoint> points(nhits);
-  vector<GlobalError> errors(nhits);
-  vector<bool> isBarrel(nhits);
-  
-  if (theTrackerWatcher.check(es)) {
-    edm::ESHandle<TrackerGeometry> trackerESH;
-    es.get<TrackerDigiGeometryRecord>().get(trackerESH);
-    theTracker = trackerESH.product();
-  }
-
-  if (theFieldWatcher.check(es)) {
+  if (!theField) { 
     edm::ESHandle<MagneticField> fieldESH;
     es.get<IdealMagneticFieldRecord>().get(fieldESH);
     theField = fieldESH.product();
```

Comment:
this can lead to segfaults.
(as an example, we had ecal geometry pointer move on an IOV boundary in prompt reco without a change in the objects payload; not obvious that magfield is protected; clearly, this will crash when spanning 0T - 3.8T mix).
IIRC, es.get should be fast anyways and the watchers/checkers are not needed (well, maybe not in HLT called per tracking region).
Better yet, percolate this from the calling producer module.